mono/utils/mono-filemap.c

```diff
@@ -57,7 +57,7 @@ int
 mono_file_map_fd (MonoFileMap *fmap)
 {
 #ifdef WIN32
-	return fileno ((FILE*)fmap);
+	return _fileno ((FILE*)fmap);
```

Comment:
These are good changes but they kinda shouldn't be needed. The underscore functions are just in oldnames.lib and aliases. There has never been any extra code supporting them so not much point in removing them either. They are a static link feature, with zero code, just name equivalences. But ok.

Reply:
This caused linker errors on Xbox. I don't think oldnames.lib exists there.

Reply:
Probably SDK should have just shipped it for developer convenience but ok.

Reply:
We have solve this using include header to get same behavior on platforms lacking oldnames.lib to not introducing additional #ifdef's (in cases when not already #ifdef:ed) around POSIX's and ISO C++ compliance differences.

Reply:
We're not adding a new `#ifdef` here, so I think this is acceptable.